### DISCUSSION TITLE
Ctrl + L does not work on SSH sessions

### DISCUSSION THREAD

**Discussion by 3f6a** [ghostty-org/ghostty#3796] [Feature Requests, Ideas] [answered]
Hi, I'm trying ghostty for the first time. After installing, I tried `ssh`'ing to some remote servers I use, and I noticed that pressing `Ctrl + L` does not clear the previous commands as usual. Is this working normally for anyone else?

`Ctrl + L` works fine if I am a local terminal session in ghostty .

Some details about my setup: I am running macOS locally, while the remote is a Linux box with OhMyZsh installed.

**Comment by jcollie** [comment, marked as answer]
https://ghostty.org/docs/help/terminfo

**Comment by 3f6a** [comment]
Not sure what I'm supposed to do with that.

Note that `Ctrl + L` usually works fine in the same scenario, with other terminal apps (such as Terminal.app or iTerm2)

**Comment by jmckillican** [comment]
do what it says to fix your terminfo on ssh'd sessions.  

do:
`infocmp -x | ssh YOUR-SERVER -- tic -x -`

or edit your ssh .config specifying the term:
```
# .ssh/config
Host example.com
  SetEnv TERM=xterm-256color
```

**Reply by 3f6a** [reply]
Thanks. The first approach
```
infocmp -x | ssh YOUR-SERVER -- tic -x -
```
didn't work. I got the warning:
```
"<stdin>", line 2, col 31, terminal 'xterm-ghostty': older tic versions may treat the description field as an alias
```
mentioned in the docs, but also the ssh connection just failed, without any error printed.

The second approach worked for me:

```
# .ssh/config
Host example.com
  SetEnv TERM=xterm-256color
```

But there is this warning in the Ghostty docs:

> Fallback does not support advanced terminal features. Because xterm-256color does not include all of Ghostty's capabilities, terminal features beyond xterm's like colored and styled underlines will not work.

which is unfortunate. I mostly use my terminal to ssh to remote servers, and the reason to install Ghostty in the first place is so I can try some of its niceties. It seems however that because of this `xterm-256color` issue, I will be missing out on most.

**Reply by Ddfulton** [reply]
Are you sure `infocmp -x | ssh <YOUR_SERVER> -- tic -x -` didn't work? I ran it, got the same "error" (which I think was actually just a warning), and now I'm off to the races, hopefully with all of the Ghostty advanced features! I'd remove that line in your `config` and see if it still works...good luck.

**Reply by peterjc** [reply]
Note that `infocmp -x | ssh <YOUR_SERVER> -- tic -x -` will return you to your terminal on the local machine. It just runs the `tic` command on your server, giving it the data from running `infocmp`` on your local machine (which gives a harmless warning as per the Ghostty docs). It will not leave you logged into the remote server.
